app/src/main/java/com/nutomic/syncthingandroid/service/EventProcessor.java

```diff
@@ -242,10 +209,82 @@ public void stop() {
         }
     }
 
-    private void notify(String text, PendingIntent pi) {
-        // HACK: Use a random, deterministic ID between 1000 and 2000 to avoid duplicate
-        //       notifications.
-        int notificationId = 1000 + text.hashCode() % 1000;
-        mNotificationHandler.showEventNotification(text, pi, notificationId);
+    private void onDeviceRejected(String deviceId, String deviceName) {
+        if (deviceId == null) {
+            return;
+        }
+        Log.d(TAG, "Unknown device " + deviceName + "(" + deviceId + ") wants to connect");
+
+        String title = mContext.getString(R.string.device_rejected,
+                deviceName.isEmpty() ? deviceId.substring(0, 7) : deviceName);
+        int notificationId = mNotificationHandler.getNotificationIdFromText(title);
+
+        // Prepare "accept" action.
+        Intent intentAccept = new Intent(mContext, DeviceActivity.class)
+                .putExtra(DeviceActivity.EXTRA_NOTIFICATION_ID, notificationId)
+                .putExtra(DeviceActivity.EXTRA_IS_CREATE, true)
+                .putExtra(DeviceActivity.EXTRA_DEVICE_ID, deviceId)
+                .putExtra(DeviceActivity.EXTRA_DEVICE_NAME, deviceName);
+        // Use a deterministic ID to make multiple PendingIntents distinguishable.
+        int requestCode = deviceId.hashCode();
```

Comment:
Why is this not the same as notificationId then?

Reply:
It was taken over from the previous code. I see no point why my comment from above would not apply here so I'll make a commit to use the notificationId (already generated above) as the requestCode. Values between 1000 and 2000 are okay here.